dist/angular-actioncable.js

```diff
@@ -138,44 +140,67 @@ ngActionCable.factory('ActionCableController', function (ActionCableConfig) {
     post: function(message){
       return route(message);
     },
-    actions: actions
+    actions: actions,
+    after_ping_callback: function(){}
   };
 
   return methods;
 });
 
-// ngActionCableSocketWrangler to start, stop or try reconnect websockets every intervalTime milliseconds.
+// ngActionCableSocketWrangler to start, stop or try reconnect websockets if they die.
 //
 // Current status is denoted by three booleans:
 // connected(), connecting(), and disconnected(), in an abstraction
 // of the internal trivalent logic. Exactly one will be true at all times.
 //
 // Actions are start() and stop()
-ngActionCable.factory("ActionCableSocketWrangler", function(ActionCableWebsocket, ActionCableConfig) {
-  var intervalTime= 8647;
+ngActionCable.factory("ActionCableSocketWrangler", function(ActionCableWebsocket, ActionCableConfig, ActionCableController) {
+  var reconnectIntervalTime= 7537;
+  var timeoutTime= 20143;
   var websocket= ActionCableWebsocket;
+  var controller= ActionCableController;
   var _live= false;
   var _connecting= false;
+  var _reconnectTimeout= false;
+  var setReconnectTimeout= function(){
+    stopReconnectTimeout();
+    _reconnectTimeout = _reconnectTimeout || setTimeout(function(){
+      if (ActionCableConfig.debug) console.log("ActionCable connection might be dead; no pings received recently");
+      connection_dead();
+    }, timeoutTime + Math.floor(Math.random() * timeoutTime / 5));
```

Comment:
This is mitigating a quirk I've seen on a system where if you restart the server or something, all clients will attempt to reconnect at the same moment.  This adds a 20% random variance to all the client timings.